plugins/caddy/main.go

```diff
@@ -20,36 +20,50 @@ import (
 )
 
 type key string
+
 const getterContextCtxKey key = "getter_context"
+const moduleName = "souin_cache"
+const moduleID caddy.ModuleID = "http.handlers." + moduleName
 
 func init() {
 	caddy.RegisterModule(SouinCaddyPlugin{})
-	httpcaddyfile.RegisterGlobalOption("souin", parseCaddyfileGlobalOption)
-	httpcaddyfile.RegisterHandlerDirective("souin", parseCaddyfileHandlerDirective)
+	httpcaddyfile.RegisterGlobalOption(moduleName, parseCaddyfileGlobalOption)
+	httpcaddyfile.RegisterHandlerDirective(moduleName, parseCaddyfileHandlerDirective)
 }
 
-var staticConfig Configuration
+var (
+	staticConfig Configuration
+	appCounter   = 0
+	appConfigs   *caddy.UsagePool
+)
 
 // SouinCaddyPlugin declaration.
 type SouinCaddyPlugin struct {
 	plugins.SouinBasePlugin
-	configuration *Configuration
+	Configuration *Configuration
 	logger        *zap.Logger
+	LogLevel      string `json:"log_level,omitempty"`
 	bufPool       sync.Pool
+	Headers       []string
+	Olric         configurationtypes.CacheProvider
+	TTL           string
```

Comment:
You're missing `json` tags for these. It probably still works because Golang automatically maps struct fields to their lowercase names for JSON keys, but it's best-practice to always include them. You should also specify `omitempty` to make sure they aren't required to specify.

Reply:
Yes it's auto mapped by golang but okay to explicit json tags